src/containers/PageContainer.jsx

```diff
@@ -96,9 +118,9 @@ class PageContainer extends React.PureComponent {
             widgets,
             tables,
             images,
-            questions,
             answers,
             shared,
+            questions,
```

Comment:
this is another sticking point.  So you were running your getQuestionNumbers method inline here.  It's important to remember that any methods in the render method run every time the render method runs.  In this case that's quite often because it is responding to changes on the page like new/modified answers, and even questions getting their active state applied, and all manner of other stuff.  We couldn't expect our question numbers to change, so we should set once and forget about it.  So I added questions to this component's local state and do just that in the componentDidMount.